spring-statemachine-uml/src/test/java/org/springframework/statemachine/plantuml/PlantUmlWriterTest.java

```diff
@@ -0,0 +1,325 @@
+                Arguments.of("org/springframework/statemachine/uml/broken-model-shadowentries", null),
+                Arguments.of("org/springframework/statemachine/uml/initial-actions", null),
+                Arguments.of("org/springframework/statemachine/uml/missingname-choice", null),
+                Arguments.of("org/springframework/statemachine/uml/multijoin-forkjoin", null),
+                Arguments.of("org/springframework/statemachine/uml/pseudostate-in-submachine", null),
+                Arguments.of("org/springframework/statemachine/uml/pseudostate-in-submachineref", null),
+                Arguments.of("org/springframework/statemachine/uml/simple-actions", null),
+                Arguments.of("org/springframework/statemachine/uml/simple-choice", null),
+                // It seems Spring statemachine does not support org.eclipse.uml2.uml.ConnectionPointReference !
+                // Arguments.of("org/springframework/statemachine/uml/simple-connectionpointref", null),
+                Arguments.of("org/springframework/statemachine/uml/simple-entryexit", null),
+                Arguments.of("org/springframework/statemachine/uml/simple-eventdefer", null),
+                Arguments.of("org/springframework/statemachine/uml/simple-flat-end", null),
+                Arguments.of("org/springframework/statemachine/uml/simple-flat-multiple-to-end-viachoices", null),
+                Arguments.of("org/springframework/statemachine/uml/simple-flat-multiple-to-end", null),
+                Arguments.of("org/springframework/statemachine/uml/simple-flat", null),
+                Arguments.of("org/springframework/statemachine/uml/simple-forkjoin",
+                        // add hidden transition to make diagram more readable
+                        new PlantUmlWriterParameters<String>()
+                                .hiddenTransition("S1", RIGHT, "S2")
+                ),
+                Arguments.of("org/springframework/statemachine/uml/simple-guards", null),
+                Arguments.of("org/springframework/statemachine/uml/simple-history-deep", null),
+                Arguments.of("org/springframework/statemachine/uml/simple-history-default", null),
+                Arguments.of("org/springframework/statemachine/uml/simple-history-shallow", null),
+                Arguments.of("org/springframework/statemachine/uml/simple-junction", null),
+                Arguments.of("org/springframework/statemachine/uml/simple-localtransition",
+                        // add some parameters to make diagram more readable
+                        new PlantUmlWriterParameters<String>()
+                                .arrowDirection("S22", UP, "S2")
+                                .arrowDirection("S21", UP, "S2")
+                                .arrowDirection("S22", UP, "S2")
+                                .arrowDirection("S21", UP, "S2")
```

Comment:
PlantUmlWriterParameters allow to tweak diagram by 'forcing' direction of some arrows / transitions